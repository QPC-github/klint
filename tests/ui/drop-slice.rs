#![crate_type = "lib"]

struct LockOnDrop;

impl Drop for LockOnDrop {
    #[klint::preempt_count(adjust = 1, unchecked)]
    fn drop(&mut self) {}
}

struct SleepOnDrop;

impl Drop for SleepOnDrop {
    #[klint::preempt_count(expect = 0)]
    fn drop(&mut self) {}
}

fn drop_lock(x: Box<[LockOnDrop]>) {}

#[klint::report_preempt_count]
fn drop_sleep(x: Box<[SleepOnDrop]>) {}
