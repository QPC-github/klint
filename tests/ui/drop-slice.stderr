WARN klint::ctxt no klint metadata found for crate core
error: dropping element of slice causes non-zero preemption count adjustment
  |
  = note: adjustment for dropping `LockOnDrop` is 1
  = note: because slice can contain variable number of elements, adjustment for dropping the slice cannot be computed statically

WARN klint::ctxt no klint metadata found for crate alloc
note: reporting preemption count for instance `drop_sleep`
  --> $DIR/drop-slice.rs:20:1
   |
20 | fn drop_sleep(x: Box<[SleepOnDrop]>) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: adjustment is inferred to be 0

note: reporting preemption count for instance `drop_sleep`
  --> $DIR/drop-slice.rs:20:1
   |
20 | fn drop_sleep(x: Box<[SleepOnDrop]>) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: expectation is inferred to be 0

error: aborting due to previous error

